pkg/networkservice/connectioncontext/dnscontext/vl3dns/server.go

```diff
@@ -38,16 +41,22 @@ import (
 	"github.com/networkservicemesh/sdk/pkg/tools/dnsutils/noloop"
 	"github.com/networkservicemesh/sdk/pkg/tools/dnsutils/norecursion"
 	"github.com/networkservicemesh/sdk/pkg/tools/ippool"
+	"github.com/networkservicemesh/sdk/pkg/tools/log"
 )
 
 type vl3DNSServer struct {
+	chanCtx               context.Context
 	dnsServerRecords      memory.Map
 	dnsConfigs            *dnsconfig.Map
 	domainSchemeTemplates []*template.Template
 	dnsPort               int
 	dnsServer             dnsutils.Handler
 	listenAndServeDNS     func(ctx context.Context, handler dnsutils.Handler, listenOn string)
-	getDNSServerIP        func() net.IP
+	dnsServerIP           net.IP
+	serverAddressCh       <-chan net.IP
+	connections           map[string]*networkservice.Connection
```

Comment:
We should get connections from the monitor server chain element. I think it's not a good solution to add a new map that duplicates connections from the monitor chain element.

Reply:
@glazychev-art Thoughts?

Reply:
I agree, it would be great to use `monitor` to get connections